src/aggregation/contribution_tracker.c

```diff
@@ -156,6 +156,10 @@ void contribution_tracker_update_aid(ContributionTrackerState *state, aid_t aid)
     state->aid_seed ^= aid;
     entry->has_contribution = false;
     state->distinct_contributors++;
+
+    add_top_contributor(&state->contribution_descriptor,
+                        &state->top_contributors,
+                        entry->contributor);
```

Comment:
>I think this is the source of the crash (and not https://github.com/diffix/pg_diffix/pull/351).

Yes, you're absolutely right, I mixed up the commits. I'll fix in the original comment.

> It doesn't make sense to me to include the entry in top contributors, since the contribution value is missing.

I have included this, because from what I've seen in the code and in the tests, `reference` does include it. The situation is when the "counted value" is `NULL`. In `reference`:
```
  interface IAggregator with
    member this.Transition args =
      match args with
...
      | [ aidInstances; Null ] -> updateAidMaps aidInstances 0L
...
```
This is the `reference`'s counterpart of `contribution_tracker_update_aid`. `aidInstances` get updated with a `0` contribution and later take part in the entire `topContributors` flow.